http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamServiceInterp.scala

```diff
@@ -211,7 +217,7 @@ class SamServiceInterp[F[_]](apiClientProvider: SamApiClientProvider[F],
                               samResourceId: SamResourceId,
                               projectParent: Option[GoogleProject],
                               workspaceParent: Option[WorkspaceId],
-                              creator: Option[WorkbenchEmail]
+                              policies: Map[String, SamPolicyData]
```

Comment:
Does it make sense to type the key here as `SamPolicyName`?

Reply:
I went with String because I don't love how `SamPolicyName` is an enum in Leo. It's not really an enumeration, this is just an arbitrary name for the policy. (Often role names are reused for policy names, see [WSM for example](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/iam/SamService.java#L1173)).

`SamPolicyName` used all over the place in the _old_ auth layer (and conflated with roles somewhat) so we can't remove/change it yet.

Reply:
That's a great reason not to do it, lol. Could you add a comment suggesting that we might want to move to a dedicated policyName type?

Reply:
Sure 👍